rclc/src/rclc/timer.c

```diff
@@ -32,15 +32,16 @@ rclc_timer_init_default(
     support, "support is a null pointer", return RCL_RET_INVALID_ARGUMENT);
 
   (*timer) = rcl_get_zero_initialized_timer();
-  rcl_ret_t rc = rcl_timer_init(
+  rcl_ret_t rc = rcl_timer_init2(
     timer,
     &support->clock,
     &support->context,
     timeout_ns,
     callback,
-    (*support->allocator));
+    (*support->allocator),
+    true);
```

Comment:
if that is okay with `rclc` not to make it available for user, this can be always true.

Reply:
Thanks, according to your comments, a `bool` argument autostart should be passed to  `rclc_timer_init_default` by user, and according to the https://github.com/ros2/rcl/blob/b55c41b8a877508f69a39daf8f0f8d9e551c69e8/rcl/include/rcl/timer.h#L172, the default use of the is argument is `true`. am i right ?

Reply:
technically it is the C impl so that user application should specify the `autostart` flag explicitly. but in `rcl_timer_init` falls back to `true` to keep the same behavior.

> a bool argument autostart should be passed to rclc_timer_init_default by user

that was the original question, i am not sure if `rclc` application needs to specify or not. if there are no use case, it can always fall back to `true` in `rclc`. but if it can be optional for `rclc`, it would be flexible for user application.